src/app-layer-parser.c

```diff
@@ -130,9 +130,15 @@ struct AppLayerParserState_ {
 
     /* State version, incremented for each update.  Can wrap around. */
     uint8_t version;
+
+    /* Indicates which loggers that have already logged the transaction,
+       using the logger id (power of two) as a bitwise flag. */
+    uint32_t logged;
```

Comment:
in this implementation the flags are store in AppLayerParserState which a per-flow structure, not per tx. This means that logging TX 1 would disable logging for TX 2. The logged field should be stored in the tx instead I think.